docs/design/ovs-pipeline.md

```diff
@@ -217,78 +306,166 @@ on the same Node and they have the following IP addresses:
-gateway or local Pod) the packet belongs to. This is done by matching on the
-ingress port for the packet. The appropriate value is then written to bits
-[0..3] in NXM_NX_REG0: 0 for tunnel, 1 for local gateway and 2 for local Pod.
-This information is used by matches in subsequent tables. For a packet received
-from the tunnel port, bit 19 in NXM_NX_REG0 is set to 1, to indicate MAC rewrite
-should be performed for the packet in [L3ForwardingTable].
+Stage is used to group tables which implement similar functions in a pipeline. At this 
+moment, we have the following stages:
 
-If you dump the flows for this table, you may see the following:
+- stageStart: to initialize PipelineRootClassifierTable.
+- stageClassifier: to classify packets "category" (tunnel, local gateway or local Pod,
+  etc).
+- stageValidation: to validate packets.
+- stageConntrackState: to transform committed packets in CT zones.
+- stagePreRouting: similar to PREROUTING chain of nat table in iptables. DNAT for
+  Service connections is performed in this stage.
+- stageEgressSecurity: to install egress rules for K8s NetworkPolicy and Antrea
+  NetworkPolicy.
+- stageRouting: L3 Forwarding of packets.
+- stagePostRouting: similar to POSTROUTING chain of nat table in iptables. SNAT for
+  Service connections is performed in this stage.
+- stageSwitching: L2 Forwarding of packets.
+- stageIngressSecurity: to install ingress rules for K8s NetworkPolicy and Antrea
+  NetworkPolicy.
+- stageConntrack: to commit non-Service connections.
+- stageOutput: to output packets to target port.
+
+### Feature
+
+Feature is the interface to program a major function in Antrea data path. At this
+moment, we have the following features:
+
+- featurePodConnectivity: to implement connectivity for Pods.
+- featureNetworkPolicy, to implement K8s NetworkPolicy and Antrea NetworkPolicy.
+- featureService, to implement K8s Service.
+- featureEgress, to implement Egress.
+- featureMulticast, to implement multicast.
+- featureTraceflow, to implement Traceflow.
+
+### Pipeline
+
+Pipeline is used to implement a major function in Antrea data path. At this moment,
+we have the following pipelines:
+
+- pipelineRoot: to classify packets to pipelineARP / pipelineIP.
+- pipelineARP: to process ARP packets.
+- pipelineIP: to process IPv4/IPv6 packets.
+- pipelineMulticast: to process multicast packets.
+
+### Table
+
+Table in Flexible Pipeline is the basic unit to build OVS pipelines. A Table can
+be referenced by one or more features, but its member struct ofTable will be
+initialized and realized on OVS only when it is referenced by any activated
+features.
+
+Flows of a flow table can vary under different conditions of operating systems,
+traffic modes, feature gates and flags. For a table, there are common flows for
+all conditions, and there are also special flows for a special condition.
+
+## Flow Tables
+
+![OVS pipeline](../assets/ovs-pipeline-antrea-proxy-1.svg)
```

Comment:
Could you refer PR #3756 to update the svg background, it helps to show diagram clearly when browser is in dark mode. thanks.